lib/opencpu/delayed_calculation.rb

```diff
@@ -50,8 +51,53 @@ def info
       process_resource @available_resources[:info].to_s
     end
 
+    def keys
+      @available_resources.keys
+    end
+
+    # available_resources(:"R/foo")
+    def available_resources(key)
+      raise ResponseNotAvailableError unless @available_resources.has_key?(key)
+      process_resource @available_resources[key].to_s
+    end
+
+    def process_resources(key, options = {})
```

Comment:
> This needs some comments,

I didn't add any comment because the codes do not contain comments in general.

> some text in the readme

I can add usage examples/scenarios in the README.

> and some specs.

I'm not very good at testing/specs, but I can still try to add.

> So it's a nested resources type thing?

It can certainly be done much better than I have offered.

However, I wanted to share with you the code that I have customized for my own needs and that I have solved.

> There also also a lot of duplication, especially in all the private methods below. We should either reuse the client here to make the request or separate the logic into a separate file.
> (Also, I think we can use named parameters now instead of an options hash. 5 years ago, we were asked to support 1.9, but now no-one should use a ruby version that has been unsupported for 4 years. but this is irrelevant if we reuse the client code)

If this GEM will be re-designed or written, there will be much to change. Yes the codes repeated and bothers me frankly, I have little time for this job. I offer a work-around solution (saves the days).

Reply:
Thing is, that we only ever use the execute endpoint and never the prepare one, so we are happy to merge useful features, but we only develop what we need ourselves.